irksome/scheme.py

```diff
@@ -49,16 +49,46 @@ class ContinuousPetrovGalerkinScheme(GalerkinScheme):
     def __init__(self, order,
                  basis_type=None,
                  quadrature_degree=None,
-                 quadrature_scheme="default"):
+                 quadrature_scheme=None):
         assert order >= 1, f"{type(self).__name__} must have order >= 1"
         super().__init__(order, basis_type=basis_type,
                          quadrature_degree=quadrature_degree,
                          quadrature_scheme=quadrature_scheme)
 
 
-def create_time_quadrature(degree, scheme="default"):
+class GalerkinCollocationScheme(ContinuousPetrovGalerkinScheme):
+    """Class for describing collocation cPG-in-time methods"""
+    def __init__(self, order,
+                 stage_type="value",
+                 quadrature_degree=None,
+                 quadrature_scheme=None):
+        assert order >= 1, f"{type(self).__name__} must have order >= 1"
+        if quadrature_scheme is None:
+            test_type = "spectral"
+        elif quadrature_scheme in {"radau", "lobatto"}:
+            test_type = quadrature_scheme
+            if quadrature_degree is None and quadrature_scheme == "lobatto":
+                # Default to under-integration
+                quadrature_degree = 2*order-2
+        else:
+            raise ValueError(f"Unsupported quadrature scheme {quadrature_scheme}.")
+
+        if stage_type not in {"deriv", "value"}:
+            raise ValueError(f"Unsupported stage type {stage_type}.")
+
+        trial_type = stage_type
+        basis_type = (trial_type, test_type)
+        super().__init__(order, basis_type=basis_type,
+                         quadrature_degree=quadrature_degree,
+                         quadrature_scheme=quadrature_scheme)
+
+
+def create_time_quadrature(degree, scheme=None):
```

Comment:
It might be clearer to use `num_points` rather than degree and then instantiate the Radau (or Gauss) with that number of points, independent of the degree we're after.  That, rather than order of accuracy, is the main thing in the structure here.